src/sources/http_scrape/tests.rs

```diff
@@ -0,0 +1,232 @@
+use std::collections::HashMap;
+use tokio::time::Duration;
+use warp::Filter;
+
+use crate::{serde::default_decoding, serde::default_framing_message_based};
+use codecs::decoding::{
+    CharacterDelimitedDecoderOptions, DeserializerConfig, FramingConfig,
+    NewlineDelimitedDecoderOptions,
+};
+use vector_core::event::Event;
+
+use super::HttpScrapeConfig;
+use crate::test_util::{
+    components::{
+        run_and_assert_source_compliance, run_and_assert_source_error, HTTP_PULL_SOURCE_TAGS,
+        SOURCE_ERROR_TAGS,
+    },
+    next_addr, test_generate_config,
+};
+
+pub(crate) const INTERVAL_SECS: u64 = 1;
+
+/// The happy path should yield at least one event and must emit the required internal events for sources.
+pub(crate) async fn run_compliance(config: HttpScrapeConfig) -> Vec<Event> {
+    let events =
+        run_and_assert_source_compliance(config, Duration::from_secs(1), &HTTP_PULL_SOURCE_TAGS)
+            .await;
+
+    assert!(!events.is_empty());
+
+    events
+}
+
+/// The error path should not yield any events and must emit the required error internal events.
```

Comment:
Simple function - but perhaps could be marked for later extracting into `test_utils`?

Reply:
🤔 The main reason I kept it here was because I wasn't sure if it was always the case that if the error internal metric is fired, that there no events result. If that is always the case, then yes I agree probably makes sense to put in `test_utils` with the parameters of config and timeout.

Reply:
Definitely a good down the line thing to think about 👍

Reply:
Added a comment